lib/spack/spack/build_systems/oneapi.py

```diff
@@ -62,24 +62,44 @@ def install(self, spec, prefix, installer_path=None):
             # with other install depends on the userid. For root, we
             # delete the installercache before and after install. For
             # non root we redefine the HOME environment variable.
-            if getpass.getuser() == 'root':
-                shutil.rmtree('/var/intel/installercache', ignore_errors=True)
-
             bash = Executable('bash')
-
-            # Installer writes files in ~/intel set HOME so it goes to prefix
-            bash.add_default_env('HOME', prefix)
             # Installer checks $XDG_RUNTIME_DIR/.bootstrapper_lock_file as well
             bash.add_default_env('XDG_RUNTIME_DIR',
                                  join_path(self.stage.path, 'runtime'))
 
-            bash(installer_path,
-                 '-s', '-a', '-s', '--action', 'install',
-                 '--eula', 'accept',
-                 '--install-dir', prefix)
-
-            if getpass.getuser() == 'root':
-                shutil.rmtree('/var/intel/installercache', ignore_errors=True)
+            if '2021' in str(spec.version):
+                if getpass.getuser() == 'root':
+                    if exists('/var/intel/installercache'):
+                        shutil.move('/var/intel/installercache', '/tmp/intel/installercache')
+
+                # Installer writes files in ~/intel set HOME so it goes to prefix
+                bash.add_default_env('HOME', prefix)
+
+                bash(installer_path,
+                     '-s', '-a', '-s', '--action', 'install',
+                     '--eula', 'accept',
+                     '--install-dir', prefix)
+
+                if getpass.getuser() == 'root':
+                    shutil.rmtree('/var/intel/installercache', ignore_errors=True)
+                    if exists('/tmp/intel/installercache'):
+                        shutil.move('/tmp/intel/installercache', '/var/intel/installercache')
+            else:
+                # Since 2022 release oneAPI installer supports multi-instance mode
+                # and can separate one installation from another. That means 
+                # all manipulations with installercahce is unnessesary. 
+                instance = prefix.split('/')[-1]
+                bash(installer_path,
+                     '-s', '-a', '-s', '--action', 'install',
+                     '--eula', 'accept',
+                     '--instance', instance,
+                     '--download-cache', '/tmp/intel/downloadcache',
+                     '--install-dir', prefix)
+                shutil.rmtree('/tmp/intel/downloadcache')
+                if getpass.getuser() == 'root':
```

Comment:
```suggestion
                # There is no support of customizing uninstallation process in Spack except for a global hook,
                # so there is no easy and elegant way to run the installer for proper product uninstallation.
                # Thus just remove the instance from the installer database now to avoid keeping garbage on the system.
                # This also prevents product instances installed by Spack from  uninstalling/modifying by the installer (without Spack).
                if getpass.getuser() == 'root':
```